Assets/MRTK/Core/Utilities/DebugUtilities.cs

```diff
@@ -30,27 +56,110 @@ public static void DebugAssert(bool condition)
         /// Logs an error message.
         /// </summary>
         /// <param name="message">The message to log.</param>
+        [Obsolete("Deprecated. Use LogError instead.")]
         public static void DebugLogError(string message)
         {
-            Debug.LogError(message);
+            LogError(message);
         }
 
         /// <summary>
         /// Logs a warning message.
         /// </summary>
         /// <param name="message">The message to log.</param>
+        [Obsolete("Deprecated. Use LogWarning instead.")]
         public static void DebugLogWarning(string message)
         {
-            Debug.LogWarning(message);
+            LogWarning(message);
         }
 
         /// <summary>
         /// Logs a message.
         /// </summary>
         /// <param name="message">The message to log.</param>
+        [Obsolete("Deprecated. Use Log instead.")]
         public static void DebugLog(string message)
         {
-            Debug.Log(message);
+            Log(message);
+        }
+
+        /// <summary>
+        /// Logs an error message.
+        /// </summary>
+        /// <param name="message">The message to log.</param>
+        public static void LogError(string message)
+        {
+            if (LogLevel <= LoggingLevel.Error)
+            {
+                Debug.LogError(message);
+            }
+        }
+
+        /// <summary>
+        /// Logs a warning message.
+        /// </summary>
+        /// <param name="message">The message to log.</param>
+        public static void LogWarning(string message)
+        {
+            if (LogLevel <= LoggingLevel.Warn)
+            {
+                Debug.LogWarning(message);
+            }
+        }
+
+        /// <summary>
+        /// Logs a message at the informational level.
+        /// </summary>
+        /// <param name="message">The message to log.</param>
+        public static void Log(string message)
+        {
+            if (LogLevel <= LoggingLevel.Information)
+            {
+                Debug.Log(message);
+            }
+        }
+
+        /// <summary>
+        /// Logs the given message to the Unity console and player log if verbose logging is enabled.
+        /// Note that verbose logs do not include the callstack
+        /// </summary>
+        /// <remarks>
+        /// If you are doing string concatentation or manipulation, use LogVerboseFormat
+        /// that formats the string, as that will not allocate memory when verbose logging is enabled.
+        ///
+        /// For example, don't do:
+        /// Debug.LogVerbose("This is my message: " + text);
+        ///
+        /// Do:
+        /// Debug.LogVerbose("This is my message: {0}", text);
+        /// </remarks>
+        public static void LogVerbose(string message)
```

Comment:
Is there a functional difference between LogInfo and LogVerbose? Both call Debug.Log, but LogVerbose explicitly sets it so no stack trace will be outputed to log, which seems like a strict subset of LogInfo

Reply:
Added some more comments here!